Use globals of original(wrapped) function

yuyupopo opened the pull request:
Fixes: https://github.com/airtai/faststream/issues/1492

The original issue happens because the function is wrapped with a decorator, and type annotation resolution is done in the the decorator's globals. The fix uses `func.__module__` instead of `func.__globals__`

This is because `wraps` copies the `__module__` of the wrapped function, but not the `__globals__`

Lancetnik commented:
Thank you a lot for the PR! I'll take a look, what we can do with old versions this evening

Lancetnik commented:
Also, it will be really glad, if you add a test for your case, that fails for current impl and passes with your changes

yuyupopo commented:
> Also, it will be really glad, if you add a test for your case, that fails for current impl and passes with your changes

could you briefly share where i should implement the test? can i make a tests/import_test folder and write the test there?

yuyupopo commented:
> Also, it will be really glad, if you add a test for your case, that fails for current impl and passes with your changes

I'll add the test somewhere and fix the code so it runs in 3.8 and 3.9 also.

Lancetnik commented:
> could you briefly share where i should implement the test? can i make a tests/import_test folder and write the test there?

You can just add a test to "tests/test_locals.py" case

yuyupopo commented:
> > could you briefly share where i should implement the test? can i make a tests/import_test folder and write the test there?
> 
> You can just add a test to "tests/test_locals.py" case

I've added the test in `tests/test_prebuild.py`. Also added another `tests/wrapper.py` because the require a wrapper function that exists in different module.

Lancetnik commented on fast_depends/utils.py:
```diff
@@ -81,9 +80,11 @@ def get_typed_signature(call: Callable[..., Any]) -> Tuple[inspect.Signature, An
-    # We need to get the globals of the wrapped function
-    globalns = sys.modules.get(call.__module__).__dict__
+    # We unwrap call to get the original unwrapped function
+    while hasattr(call, "__wrapped__"):
```
Is it the same with `inspect.unwrap`?

yuyupopo replied:
I think so. I didn't know about that. Since `unwrap` was added in 3.4, I'll change to that. 
Also, I forgot pydantic v1 didn't have `'model_rebuild'` yet. I'll fix this one also

yuyupopo commented:
I've addressed your comment, and this time hopefully it will pass all the tests

Lancetnik commented:
Yeah, thank you a lot for the fix!

Lancetnik commented:
> Would it be possible to release another version by the weekend?

I tried to release it yesterday already, but there are some troubles with `dirty-equals` and python3.8 tests. I have to fixed it before we can make a release

Lancetnik commented:
@yuyupopo the job is done - you can update **FastDepends** to 2.4.4 - all should works fine

yuyupopo commented:
> @yuyupopo the job is done - you can update **FastDepends** to 2.4.4 - all should works fine

Thanks a lot!

Lancetnik commented:
> > @yuyupopo the job is done - you can update **FastDepends** to 2.4.4 - all should works fine
> 
> Thanks a lot!

Thank your for the PR!
